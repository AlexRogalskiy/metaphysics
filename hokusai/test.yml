---
version: '2'
services:
  metaphysics:
    command: yarn test
    extends:
      file: common.yml
      service: metaphysics
    environment:
    - NODE_ENV=test
    env_file: ../.env.test
