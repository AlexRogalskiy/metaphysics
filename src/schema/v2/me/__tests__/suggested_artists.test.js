/* eslint-disable promise/always-return */
import { runAuthenticatedQuery } from "schema/v2/test/utils"

describe("Me", () => {
  describe("SuggestedArtists", () => {
    const context = {
      suggestedArtistsLoader: () => {
        return Promise.resolve({
          body: [
            {
              id: "andy-warhol",
              birthday: "1919",
              artworks_count: 50,
            },
          ],
        })
      },
    }

    it("returns sanitized messages", () => {
      const query = `
        {
          me {
            suggestedArtists(artistID: "andy-warhol") {
              slug
              birthday
            }
          }
        }
      `

      return runAuthenticatedQuery(query, context).then(
        ({ me: conversation }) => {
          expect(conversation).toMatchSnapshot()
        }
      )
    })
  })
})
