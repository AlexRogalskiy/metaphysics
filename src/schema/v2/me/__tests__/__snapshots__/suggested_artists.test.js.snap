// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Me SuggestedArtists returns sanitized messages 1`] = `
Object {
  "suggestedArtists": Array [
    Object {
      "birthday": "1919",
      "slug": "andy-warhol",
    },
  ],
}
`;
